File: Civitas - Java/src/civitas/SorpresaConvertirEspeculador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author juanc
 */
public class SorpresaConvertirEspeculador extends Sorpresa {
    
    SorpresaConvertirEspeculador(int valor, String texto){
        super(valor , texto);       
    }
    
    @Override
    void aplicarAJugador (int actual, ArrayList<Jugador> todos ){ 
        informe(actual, todos);
        todos.set( actual, todos.get(actual).convertirJugador(valor) ) ;
    }
    
}

File: Civitas - Ruby/lib/sorpresa_convertir_especulador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.



module Civitas
  class Sorpresa_convertir_especulador < Sorpresa
    public_class_method :new
    def initialize(valor, texto)
      super texto
      @valor = valor
    end
    
    def aplicar_a_jugador(actual, todos)  
      informe(actual, todos)
      todos[actual] = todos[actual].convertir_especulador @valor
    end
    
  end
end
